### ISSUE TITLE
"Disable telemetry" refinements

### ISSUE THREAD

**Issue by gspencergoog** [flutter/flutter#124411]
I recently updated my master branch, and noticed some things having to with the telemetry notification:

1) When the notice about telemetry collection shows in VSCode, it doesn't put anything in the output log, so once dismissed it can't be easily re-read.
2) I initially misread the command as "run (flutter|dart) disable-telemetry" instead of the actual command "run (flutter|dart) --disable-telemetry" (at least I assume I misread it: I can't go back and see).  Could we just make `disable-telemetry` an actual command (either in addition to the flag, or instead of)?
3) When it is run, it's not clear to me what telemetry it disables, or what the scope is: for example, if I disable telemetry for the `dart` command, does that also disable it for the `flutter` command, or do I have to run it once for each? It would be good if the notifications that the telemetry has been disabled included this information (and it would be nice if the messages on both Dart and Flutter were similar: right now they are not).

cc @eliasyishak

**Comment by eliasyishak** [comment]
Thank you so much for these comments about the new unified analytics implementation. I'll make sure the rest of the team sees this as well but to just quickly answer your questions, read below

1. That is the intention, the analytics can't be collected from the VS Code (language server) until the user has explicitly dismissed the popup message that comes up. Once dismissed, it won't come back again. Was there an issue with how visible it was the first time?
2. hmm that's interesting, i would imagine the message may be formatted differently based on the user's font size preferences. But as far as changing the actual command, we would need to update the privacy document if we wanted to change `--disable-telemetry` to be a standalone subcommand. I'll discuss this more with the rest of the implementation team
3. Thank you for this point, after re-reading the consent message it definitely isn't clear that when you disable telemetry it disables it for **all** flutter and dart related. Like you said, I think the best way to communicate this would be to let users know after opting out that they have indeed have opted out of everything flutter and dart related (issue for myself to fix this https://github.com/flutter/flutter/issues/124413)


cc @bwilkerson @jcollins-g @jacob314

**Comment by gspencergoog** [comment]
> Once dismissed, it won't come back again. Was there an issue with how visible it was the first time?

No, it was plenty visible, but since it's a popup notification, I dismissed it before I went to my terminal to type the command, and then didn't have any way to refer to the notification to check that the command I typed was correct (it wasn't!).  It would be nicer if the dialog just had two buttons on it "disable telemetry" and "continue with telemetry enabled" (or some such wording, preferably lawyer-approved). Then I wouldn't need to go to my terminal and remember the command and spell it correctly.

> But as far as changing the actual command, we would need to update the privacy document if we wanted to change --disable-telemetry to be a standalone subcommand.

The advantage of it being a standalone command is that it can have richer help text associated with it (because you can say `flutter help disable-telemetry`), and it might be easier to find because of that (although it is clearly in there in the options section, that part scrolls off the top on my typical terminal size).

I also just noticed that the help text for the option:
```
    --disable-telemetry     Disable telemetry reporting when this command runs.
```

Seems to imply that it will only disable the telemetry in the command that is currently being run, not that it will disable telemetry collection in general if you run it without a command. Having a separate command (with associated clearly worded output when it does run) could let you make that an actual distinction: running with the option could disable it for a particular command, and running the command could disable it for everything from then on.

**Comment by eliasyishak** [comment]
Just following up again to see if we addressed any concerns. We have updated the helper text to show the difference between the two suppress/disable commands

```
    --suppress-analytics    Suppress analytics reporting for the current CLI invocation.
    --disable-telemetry     Disable telemetry reporting when this command runs.
```

**Comment by gspencergoog** [comment]
> Disable telemetry reporting when this command runs.

Is this only for the current CLI invocation?  Or every time?  I think it's every time, and in that case, clearer text would be:

"Disable telemetry reporting each time a flutter or dart command runs, until it is re-enabled."

**Comment by gspencergoog** [comment]
Also, is the opposite flag (to re-enable) called `--no-disable-telemetry`?  Maybe `--suppress-telemetry` would be a better name for the flag, since `--no-suppress-telemetry` is less of a double-negative, which are against our style guide, and also is similar to the analytics switch.

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-double-negatives-in-apis

**Comment by eliasyishak** [comment]
Created this PR (wip) to address this:
- https://github.com/flutter/flutter/pull/125995

**Comment by eliasyishak** [comment]
Closing since we merged in #125995
